sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,182 @@ class RotatingCertificateCredential implements TokenCredential {
 
 In this example, the custom credential type `RotatingCertificateCredential` again uses a `ClientCertificateCredential` instance to retrieve tokens. However, in this case, it will attempt to refresh the certificate before obtaining the token. The method `RefreshCertificate` will query to see if the certificate has changed. If so, it will replace `this.credential` with a new instance of the certificate credential using the same certificate path.
 
+### Controlling User Interaction
+
+In many cases, applications require tight control over user interaction. In these applications, automatically blocking on required user interaction is often undesired or impractical. For this reason, credentials in the `@azure/identity` library which interact with the user offer mechanisms to fully control user interaction.These settings are available under `InteractiveCredentialOptions` in both node and browser.
+
+```
+  const options: InteractiveCredentialOptions = {
+    disableAutomaticAuthentication: true
+  };
+  const credential = new InteractiveBrowserCredential(options);
+  await credential.authenticate("https://vault.azure.net/.default");
+  const client = new SecretClient("https://key-vault-name.vault.azure.net", credential);
+```
+
+In this sample, the application is again using the `InteractiveBrowserCredential` to authenticate a SecretClient, but with two major differences from our first example. First, in this example, the application is explicitly forcing any user interaction to happen before the credential is given to the client by calling the `authenticate` method.
+
+The second difference is here the application is preventing the credential from automatically initiating user interaction. Even though the application authenticates the user before the credential is used, further interaction might still be needed, for instance, in the case that the user's refresh token expires or a specific method require additional consent or authentication.
+
+By setting the option `disableAutomaticAuthentication` to `true` the credential will fail to automatically authenticate calls where user interaction is necessary. Instead, the credential will throw an error `AuthenticationRequiredError`. The following example demonstrates an application handling such an exception to prompt the user to authenticate only after some application logic has completed.
+
+```
+try {
+    await client.getSecret("secret-name");
+} catch (e) {
+  await ensureAnimationCompleted();
```

Comment:
I think I understand where you come from when you add this, but we’ve gotten feedback in the past about undefined functions or variables on our samples. Perhaps we could use a more descriptive comment instead of this function call?

Reply:
Yeah I took this from .NET. May be I'll just remove this